### PR TITLE
Update one drive thumbnails

### PR DESCRIPTION
**Pull request by carolanitz**
### Checklist
- [x] I've run `bundle exec fastlane test` from the root directory to see all new and existing tests pass
- [x] I've followed the [vlc-ios code style](Docs/CodingStyle.md)
- [x] I've read the [Contribution Guidelines](https://github.com/videolan/vlc-ios#contribute)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

File: Sources/VLCCloudStorageTableViewCell.h [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #import <BoxSDK/BoxSDK.h>
 #if TARGET_OS_IOS
 #import "GTLRDrive.h"
+#import <OneDriveSDK/OneDriveSDK.h>
```
Comment (Mikanbu) [inline Sources/VLCCloudStorageTableViewCell.h]: I'm not sure we want to have all the headers included here.

Wouldn't it be better to have the specific ones we need?

File: Sources/VLCCloudStorageTableViewCell.m [hunk 2]
```diff
@@ -72,6 +68,22 @@ - (void)updateOneDriveDisplayAsFolder
     _thumbnailView.image = [UIImage imageNamed:@"folder"];
 }
 
+- (void)loadThumbnail
+{
+    // The dropboxapi has no way to cancel a request and the ODThumbnail has no back reference to it's item
```
Comment (Mikanbu) [inline Sources/VLCCloudStorageTableViewCell.m]: The OneDrive API? 😄

File: Sources/VLCOneDriveTableViewController.m [hunk 3]
```diff
@@ -127,7 +127,7 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
         self.title = selectedItem.name;
     } else {
         VLCMediaList *mediaList;
-        NSURL *url = [NSURL URLWithString:selectedItem.dictionaryFromItem[@"@content.downloadUrl"]];
+        NSURL *url = [NSURL URLWithString:selectedItem.webUrl];
```
Comment (Mikanbu) [inline Sources/VLCOneDriveTableViewController.m]: Have you tested this?

I remember trying this before the sad realisation that I needed to use the dictionary. 😢

File: Sources/VLCCloudStorageTableViewCell.m [hunk 4]
```diff
@@ -72,6 +68,22 @@ - (void)updateOneDriveDisplayAsFolder
...
+    // so this might lead to wrong thumbnails if the cell is reused since we have no way of cancelling requests or check if the completion is still for the set item
+    ODDriveRequestBuilder *drive = [[ODClient loadCurrentClient] drive];
+    ODThumbnailRequest *request = [[[[drive items:_oneDriveFile.id] thumbnails:@"0"] medium] request];
+    __weak typeof(self) weakSelf = self;
+    [request getWithCompletion:^(ODThumbnail *response, NSError *error) {
+        if (error == nil && response.url) {// we don't care about errors for thumbnails
```
Comment (Mikanbu) [inline Sources/VLCCloudStorageTableViewCell.m]: Should we consider logging the error if the thumbnail fails?

File: Sources/VLCCloudStorageTableViewCell.h (hunk 1 above)
Comment (carolanitz) [reply Sources/VLCCloudStorageTableViewCell.h]: @Mikanbu we do that with Google drive and Box here as well

File: Sources/VLCOneDriveTableViewController.m (hunk 3 above)
Comment (carolanitz) [reply Sources/VLCOneDriveTableViewController.m]: yes tested it

Comment (Mikanbu) [reply Sources/VLCOneDriveTableViewController.m]: This doesn't work for me.

File: Sources/VLCCloudStorageTableViewCell.h (hunk 1 above)
Comment (carolanitz) [reply Sources/VLCCloudStorageTableViewCell.h]: but you might be right

File: Sources/VLCCloudStorageTableViewCell.m (hunk 4 above)
Comment (carolanitz) [reply Sources/VLCCloudStorageTableViewCell.m]: we can do debug logging I guess

File: Sources/VLCOneDriveTableViewController.m (hunk 3 above)
Comment (carolanitz) [reply Sources/VLCOneDriveTableViewController.m]: streaming

**Comment by Mikanbu** [comment]
Merged with ba47b33d9b8d4c0293db75bc0d3d7b4e25f4b7ac and e4715ba8a121ade877b3f39291dd0773ae350a3f.

**Pull request closed without merging**